test/submit_single_test.go

```diff
@@ -0,0 +1,337 @@
+
+		const callbackNumbers = 2                                                // cannot be greater than 5
+		callbackReceivedChan := make(chan *TransactionResponse, callbackNumbers) // do not block callback server responses
+		callbackErrChan := make(chan error, callbackNumbers)
+		callbackIteration := 0
+
+		calbackResponseFn := func(w http.ResponseWriter, rc chan *TransactionResponse, ec chan error, status *TransactionResponse) {
+			callbackIteration++
+
+			// Let ARC send the callback few times. Respond with success on the last one.
+			respondWithSuccess := false
+			if callbackIteration < callbackNumbers {
+				t.Logf("%d callback received, responding bad request", callbackIteration)
+				respondWithSuccess = false
+
+			} else {
+				t.Logf("%d callback received, responding success", callbackIteration)
+				respondWithSuccess = true
+			}
+
+			err = respondToCallback(w, respondWithSuccess)
+			if err != nil {
+				t.Fatalf("Failed to respond to callback: %v", err)
+			}
+
+			rc <- status
+		}
+		callbackUrl, token, shutdown := startCallbackSrv(t, callbackReceivedChan, callbackErrChan, calbackResponseFn)
+		defer shutdown()
+
+		resp := postRequest[TransactionResponse](t, arcEndpointV1Tx, createPayload(t, TransactionRequest{RawTx: hex.EncodeToString(tx.ExtendedBytes())}),
+			map[string]string{
+				"X-WaitFor":       Status_SEEN_ON_NETWORK,
+				"X-CallbackUrl":   callbackUrl,
+				"X-CallbackToken": token,
+			}, http.StatusOK)
+
+		require.Equal(t, Status_SEEN_ON_NETWORK, resp.TxStatus)
+
+		generate(t, 10)
+
+		statusUrl := fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx.TxID())
+		statusResp := getRequest[TransactionResponse](t, statusUrl)
+
+		require.NotNil(t, statusResp.MerklePath)
+		_, err = bc.NewBUMPFromStr(*statusResp.MerklePath)
+		require.NoError(t, err)
+
+		for i := 0; i < callbackNumbers; i++ {
+			t.Logf("callback iteration %d", i)
+			callbackTimeout := time.After(time.Second * time.Duration(i) * 2 * 5)
+
+			select {
+			case callback := <-callbackReceivedChan:
+				require.NotNil(t, callback)
+
+				t.Logf("Callback %d result: %s", i, callback.TxStatus)
+				require.Equal(t, statusResp.Txid, callback.Txid)
+				require.Equal(t, *statusResp.BlockHeight, *callback.BlockHeight)
+				require.Equal(t, *statusResp.BlockHash, *callback.BlockHash)
+				require.Equal(t, Status_MINED, callback.TxStatus)
+
+			case err := <-callbackErrChan:
+				t.Fatalf("callback received - failed to parse callback %v", err)
+			case <-callbackTimeout:
+				t.Fatal("callback not received - timeout")
+			}
+		}
+	})
+}
+
+func TestSkipValidation(t *testing.T) {
+	tt := []struct {
+		name              string
+		skipFeeValidation bool
+		skipTxValidation  bool
+	}{
+		{
+			name:              "post transaction with skip fee",
+			skipFeeValidation: true,
+			skipTxValidation:  false,
+		},
+		{
+			name:              "post transaction with skip script validation",
+			skipFeeValidation: false,
+			skipTxValidation:  true,
+		},
+		{
```

Comment:
`skipFeeValidation` is redundant here.
We could add test case where we setup `X-SkipFeeValidation = false` and `X-SkipScriptValidation= false` but `X-SkipTxValidation = true` and check if invalid transaction will be accepted by ARC